spec.html

```diff
@@ -27834,14 +27834,18 @@ <h1>Math.hypot ( _value1_, _value2_, ..._values_ )</h1>
         <p>Returns the square root of the sum of squares of its arguments.</p>
         <p>When the `Math.hypot` method is called with at least two arguments _value1_ and _value2_ and any number of additional arguments which form the rest parameter ..._values_, the following steps are taken:</p>
         <emu-alg>
-          1. Let _numbers_ be a List containing _value1_, _value2_, and the elements of _values_ in List order.
+          1. Let _args_ be a List containing _value1_, _value2_, and the elements of _values_ in List order.
+          1. Let _coerced_ be a new empty List.
+          1. For each element _arg_ of _args_, do
+            1. Let _n_ be ? ToNumber(_arg_).
+            1. Append _n_ to _coerced_.
           1. Let _onlyZero_ be *true*.
-          1. For each element _number_ of _numbers_, do
-            1. Let _n_ be ? ToNumber(_number_).
-            1. If _n_ is *NaN*, _n_ is *+&infin;*, or _n_ is *-&infin;*, return _n_.
-            1. If _n_ is neither *+0* nor *-0*, set _onlyZero_ to *false*.
+          1. For each element _number_ of _coerced_, do
+            1. If _number_ is *NaN* or _number_ is *+&infin;*, return _number_.
+            1. If _number_ is *-&infin;*, return *+&infin;*.
+            1. If _number_ is neither *+0* nor *-0*, set _onlyZero_ to *false*.
```

Comment:
this now eagerly ToNumbers everything, even if the previous logic would have had it return early - I believe the ToNumbers *have* to be interleaved with the checking logic, as in your first commit.

Reply:
Nope, the checks must come after ToNumber, hence this commit. You can confirm this in implementations. The change in #2122 was incorrect (and presumably accidental).

The previous wording was

> Each of the following `Math` object functions applies the ToNumber abstract operation to each of its arguments (in left-to-right order if there is more than one). If ToNumber returns an abrupt completion, that Completion Record is immediately returned. Otherwise, the function performs a computation on the resulting Number value(s).

Reply:
aha, you're correct.